### PR TITLE
Migrate deprecated 'assert' import syntax, bump to Node 18.20

### LINKED ISSUE #35825: Undocumented requirements on Node version for building the site
Currently the docs are not complete, as building the site per the contributor docs does not work in neither Node 20 nor Node 22. I conferred the `engines` spec and it does not reflect reality:
```
❯ jq .engines package.json
{
  "node": ">=18.18.0"
}
```

Stating the working versions and/or having a Github workflow that builds the site using all LTS versions of Node would be a start in fixing this.

# Results of building

## Node 20.9.0
```
  npx yarn start
yarn run v1.22.22
$ yarn up-to-date-check && env-cmd --silent cross-env CONTENT_ROOT=files REACT_APP_DISABLE_AUTH=true BUILD_OUT_ROOT=build yar
i-server
$ node scripts/up-to-date-check.js                                                                                           ir current behavior and syntax as those might change in a future version of Node.js.
(Use `node --trace-warnings ...` to show where the warning was created)
(node:80436) ExperimentalWarning: Importing JSON modules is an experimental feature and might change at any time
file:///Users/carlerik/dev/mdn-content/node_modules/compute-baseline/dist/browser-compat-data/compat.js:1
import bcd from "@mdn/browser-compat-data" with { type: "json" };
                                           ^^^^

SyntaxError: Unexpected token 'with'
    at ModuleLoader.moduleStrategy (node:internal/modules/esm/translators:118:18)
    at callTranslator (node:internal/modules/esm/loader:273:14)
    at ModuleLoader.moduleProvider (node:internal/modules/esm/loader:278:30)

Node.js v20.9.0
error Command failed with exit code 1.
Node.js v20.9.0
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

## Node 22.8
```
  npx yarn start
yarn run v1.22.22
$ yarn up-to-date-check && env-cmd --silent cross-env CONTENT_ROOT=files REACT_APP_DISABLE_AUTH=true BUILD_OUT_ROOT=build yar
i-server
$ node scripts/up-to-date-check.js                                                                                           ir current behavior and syntax as those might change in a future version of Node.js.
(Use `node --trace-warnings ...` to show where the warning was created)
(node:80436) ExperimentalWarning: Importing JSON modules is an experimental feature and might change at any time
file:///Users/carlerik/dev/mdn-content/node_modules/compute-baseline/dist/browser-compat-data/compat.js:1
import bcd from "@mdn/browser-compat-data" with { type: "json" };
                                           ^^^^

SyntaxError: Unexpected token 'with'
    at ModuleLoader.moduleStrategy (node:internal/modules/esm/translators:118:18)
    at callTranslator (node:internal/modules/esm/loader:273:14)
    at ModuleLoader.moduleProvider (node:internal/modules/esm/loader:278:30)

Node.js v20.9.0
error Command failed with exit code 1.
Node.js v20.9.0
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

## Some analysis
I see that the [`assert` syntax failing the Node 20 build is deprecated and removed in Node 22](https://github.com/nodejs/node/issues/51622), so I can try fixing that script.

### PR DESCRIPTION
**Pull request by fatso83**
### Description
This fixes #35825 by ensuring that `yarn start` works. Currently this breaks on both Node 20 and 22 due to scripts using a mix of syntax only supported by either version.

### Motivation
I wanted to contribute a tiny fix and this made it impossible to check the result by firing up `yarn start` (it still does, by the way, as the proxy feature doesn't seem to work on my Mac). Now, at least, I can run tests and linting.

### Additional details

- **Remove deprecated 'assert' import syntax in favor of 'with'**
- **Specify engines field to minimum Node 22 for experimental 'import ...with' support**

### Related issues and pull requests

Fixes #35825

**Background**
https://github.com/nodejs/node/issues/51622

> Note that with is already unflagged in 20 and 21 (and in 18 right now it just doesn't exist, not even behind a flag).

### CODE REVIEW

**Comment by fatso83** [comment]
After seeing that this PR failed in the Github workflow due to `.nvmrc` setting the Node version to 18.20.4, I installed that locally, and that worked. I could revert the `engines` change, but still, there is no documentation on which Node version to use in the docs (not even mentioning NVM), so that does not really fix anything. Also, Node 18 is reaching EOL in these days and it's about time to update the version anyway 😄 

I'll await some feedback. I can for instance include a reference in the contributor docs to look at `.nvmrc` for the supported version.

**Comment by bsmth** [comment]
Thanks for opening this one. We have a tracker for moving to Node 20 in MDN repos here: https://github.com/mdn/mdn/issues/521

**Comment by fatso83** [comment]
According to the linked issue no one should start working on this yet. Should I then just revert all nvmrc and package.json changes, but keep the rest and explicitly set the engine to ">=18.18.0, < 19.0.0" to avoid issues such as the ones I got?

I could also make a mention in the contributing.md docs about referring to `.nvmrc` for which Node version to use?

**Comment by caugner** [comment]
> Should I then just revert all nvmrc and package.json changes, but keep the rest and explicitly set the engine to ">=18.18.0, < 19.0.0" to avoid issues such as the ones I got?

We strive to support all current and future LTS versions of Node.js, and currently also support the active LTS version 18. So we don't want to restrict the engine to `< 19.0.0`.

The good news is that Node 18.20.0 [added support for import attributes](https://nodejs.org/en/blog/release/v18.20.0#added-support-for-import-attributes), so we should require `>=18.20.0` instead and migrate the deprecated `assert` syntax.

@fatso83 Can you update your PR accordingly? 🙏

**Comment by fatso83** [comment]
Updates done

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -158,6 +158,13 @@ node -v
 yarn -v
 ```
 
+Check that the Node version at least satisfies what is set in package.json:
+
+```shell
```
Comment (caugner) [inline CONTRIBUTING.md:163]: I'm not sure this is necessary, because yarn install will error out if the engine doesn't satisfy the version range.

Comment (fatso83) [reply CONTRIBUTING.md]: removed :)

**Review by caugner — APPROVED** [review]

**Comment by caugner** [comment]
Thanks @fatso83!

**Pull request merged**